ios/browser/api/networking/models/BraveCertificateSubjectModel.mm

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import "BraveCertificateSubjectModel.h"
+#if defined(BRAVE_CORE) // Compiling in Brave-Core
+  #import "brave/ios/browser/api/networking/common/brave_certificate_enums.h"
+  #include "brave/ios/browser/api/networking/utils/brave_certificate_ios_utils.h"
+  #include "brave/ios/browser/api/networking/utils/brave_certificate_utils.h"
+#else
+  #import "brave_certificate_enums.h"
+  #include "brave_certificate_ios_utils.h"
+  #include "brave_certificate_utils.h"
+#endif
+
+@implementation BraveCertificateSubjectModel
+- (instancetype)initWithCertificate:(X509*)certificate {
+  if ((self = [super init])) {
+    X509_NAME* subject_name = X509_get_subject_name(certificate);
```

Comment:
Should there be a null / falsy check here? (ex: `if (!subject_name) return self;`). Every cert has a subject name, but I'm not sure if OpenSSL will load invalid certs and keep going (or maybe this may get hit in unit test case w/ bunk cert data)

Reply:
Not necessary as if it is null, the `x509_utils` will return empty string:

```c++
std::string name_entry_from_nid(X509_NAME* name, std::int32_t nid)
{
  if (name) {
    //...
  }
  return std::string();
}
```

But if you want, I can add it.

Reply:
nope, it's all good! Nice fixes here 😄